packages/medusa/src/services/fulfillment.ts

```diff
@@ -224,11 +234,34 @@ class FulfillmentService extends TransactionBaseService {
 
       const created = await Promise.all(
         fulfillments.map(async ({ shipping_method, items }) => {
+          await pvInventoryServiceTx.validateInventoryAtLocation(
+            items,
+            locationId!
```

Comment:
suggestion: add a guard above that skips these steps if no `locationId` is provided

Reply:
including the promise below